### DISCUSSION TITLE
Use native Mac open & save file dialoges

### DISCUSSION THREAD

**Discussion by peterstev** [freeplane/freeplane#1398] [Ideas]
As a Mac user, the only challenge I have using FreePlane are the file manager dialogues.

Unlike other Mac apps, Freeplane does not offer the shortcuts list nor the recently used directories that are typical  of Mac applications. This makes it very hard to find things.

I am sure I am not the first ask for this (though my search did find anything. My bad if it was already there), but this would be really nice to have.

I have attached screen shots of Freeplane and MacVim getting ready to save a file.

Thank you in advance!

![Freeplane-save](https://github.com/freeplane/freeplane/assets/144412931/2cf5f130-2112-40be-8aee-f61fa4a25dad)
![MacVim-Save](https://github.com/freeplane/freeplane/assets/144412931/8d3d3a2e-43eb-4fa8-b320-ac76dd1e37d8)

**Comment by Foxxsayy** [comment]
Doing this for Windows would be great as well. There's no way to copy/paste a URL, so you have to navigate from pretty close to the root directory a lot.

**Reply by euu2021** [reply]
It was discussed in the past (in the old forum, I think). If I remember correctly, it's technically difficult to implement.

**Reply by peterstev** [reply]
Just because it is technically difficult, does not mean it is not the right thing to do :-) Consider Jakob's Law: Users spend most of their time in other apps. Users will prefer your app if it is similar to other apps.

**Comment by beccon4** [comment]
The same goes for Linux - Gnome / KDE / Xfce. Would it be possible to integrate the Freedesktop Specifications

https://www.freedesktop.org/wiki/Specifications/

to be able to use Freeplane like any other program e.g. LibreOffice?

My issue is that I have all my files on different NAS shares (CIFS, SSH, WebDav) but I'm unable to open them from within Freeplane without workarounds. (I'm probably not the only Linux user)
